vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -300,9 +300,6 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
 
             const parentItemContext = dataProviderController.getItemContext(params.parentItem);
             if (cache[parentUniqueKey] && cache[parentUniqueKey][page] && parentItemContext.subCache) {
-              // workaround: sometimes grid-element gives page index that overflows
-              page = Math.min(page, Math.floor(cache[parentUniqueKey].size / grid.pageSize));
```

Comment:
(a) The web component itself can't request pages beyond the size since it creates requests synchronously with the rows iteration, (b) with the workaround, it's possible that a callback receives items of a different page, which is incorrect.